### ISSUE TITLE
Multi-core untilize does not support tensors with large width dimensions

### ISSUE THREAD

**Issue by skhorasganiTT** [tenstorrent/tt-metal#9022]
Currently, multi-core untilize will exceed the max l1 size when operating on tensors with large width dims. This is required for Falcon7/40B which are currently using single-core untilize on the output logits (with dims up to [1,1,2048,65024]), resulting in a device perf bottleneck. 

Example test:
```
import torch
import pytest
import tt_lib
from models.utility_functions import torch2tt_tensor, tt2torch_tensor
from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import (
    get_atol_rtol_pcc,
)

@pytest.mark.parametrize("multicore", [False, True])
@pytest.mark.parametrize("expected_pcc", [1.0])
def test_untilize(multicore, expected_pcc, device):
    tensor_host = torch.randn(1,1,2048,65024, dtype=torch.bfloat16)
    tt_tensor_device = torch2tt_tensor(tensor_host, device)
    tt_tensor_device = tt_lib.tensor.untilize(tt_tensor_device, use_multicore=multicore)
    tt_tensor_host = tt2torch_tensor(tt_tensor_device)
    _, _, device_pcc, _  = get_atol_rtol_pcc(tensor_host, tt_tensor_host)
    print("PCC:", device_pcc)
    assert device_pcc >= expected_pcc
```

**Comment by skhorasganiTT** [comment]
Hi @yan-zaretskiy, is there any update on this issue?

**Comment by uaydonat** [comment]
see https://github.com/tenstorrent/tt-metal/issues/11509
